clang-tools-extra/clang-tidy/performance/UnnecessaryValueParamCheck.cpp

```diff
@@ -62,6 +61,30 @@ bool isExplicitTemplateSpecialization(const FunctionDecl &Function) {
   return false;
 }
 
+bool isPassedToStdMove(const ParmVarDecl &Param, ASTContext &Context) {
+  const auto paramName = Param.getName();
+  // Check if the parameter has a name, in case of functions like -
+  // void func(const ExpensiveToCopyType)
+  // {
+  //
+  // }
+  // The function having an empty body will still have a FunctionDecl and its
+  // parmVarDecl picked up by this checker. It will be an empty string and will
+  // lead to an assertion failure when using hasName(std::string) being used
+  // in the matcher below. If empty then exit indicating no move calls present
+  // for the function argument being examined.
+  if (paramName.empty()) {
+    return false;
+  }
+  auto Matches = match(
+      callExpr(
+          callee(functionDecl(hasName("::std::move"))), argumentCountIs(1),
```

Comment:
Should argument count only be one here? What about situations where there's more than one move call such that the variable could be recreated and moved again?

Reply:
The argument count here refers to the number of arguments in a function call. Since `std::move()` has only one argument passed to it, therefore it has been set to 1. The same concept has been applied to the move call matcher in `performance-move-const-arg` here - https://github.com/Esri/llvm-project/blob/runtimecore/clang-tools-extra/clang-tidy/performance/MoveConstArgCheck.cpp#L47